src/plugins/intel_cpu/src/nodes/executors/dnnl/dnnl_fullyconnected_primitive.cpp

```diff
@@ -0,0 +1,356 @@
+                                                                  const PostOps& postOps,
+                                                                  const MemoryArgs& memory,
+                                                                  const ExecutorContext::CPtr context,
+                                                                  const bool cacheWeights) {
+    DEBUG_LOG("Creating shape agnostic data");
+    auto srcDesc = memory.at(ARG_SRC)->getDescPtr();
+    const auto& weiDesc = memory.at(ARG_WEI)->getDescPtr();
+    auto dstDesc = memory.at(ARG_DST)->getDescPtr();
+
+    if (srcDesc->getShape().isDynamic()) {
+        const auto& inShape = srcDesc->getShape();
+        const auto& wShape = weiDesc->getShape();
+        const auto& inDymmyDims = makeDummyInputDims(inShape, wShape);
+        srcDesc = srcDesc->cloneWithNewDims(inDymmyDims);
+        const auto& outDymmyDims =
+            makeDummyOutputDims(inDymmyDims, wShape.getStaticDims(), dstDesc->getShape().getRank());
+        dstDesc = dstDesc->cloneWithNewDims(outDymmyDims);
+    }
+
+    const dnnl::memory::desc srcDnnlDesc = MemoryDescUtils::convertToDnnlMemoryDesc(srcDesc)->getDnnlDesc();
+    const dnnl::memory::desc weiDnnlDesc = MemoryDescUtils::convertToDnnlMemoryDesc(weiDesc)->getDnnlDesc();
+    const dnnl::memory::desc dstDnnlDesc = MemoryDescUtils::convertToDnnlMemoryDesc(dstDesc)->getDnnlDesc();
+
+    const auto postOpData = createPrimitiveAttrs(attrs, postOps, memory, context);
+
+    if (!cacheWeights)
+        return std::make_shared<DnnlShapeAgnosticData>(postOpData);
+
+    const auto useSparseWeights = attrs.sparseWeights;
+    const auto useWeightsDecompression = useWeightsDecompressionImpl(srcDesc->getPrecision(), weiDesc->getPrecision());
+
+    const dnnl::memory::desc biaDnnlDesc =
+        memory.at(ARG_SRC)->getDescPtr()->getCurrentMemSize() != 0
+            ? MemoryDescUtils::convertToDnnlMemoryDesc(memory.at(ARG_SRC)->getDescPtr())->getDnnlDesc()
+            : dnnl::memory::desc{};
+
+    const auto primDesc = createPrimitiveDesc(srcDnnlDesc,
+                                              weiDnnlDesc,
+                                              biaDnnlDesc,
+                                              dstDnnlDesc,
+                                              postOpData.attr,
+                                              context->getEngine(),
+                                              context->getImplPriorities(),
+                                              useSparseWeights,
+                                              useWeightsDecompression);
+
+    const auto weightsDesc = DnnlExtensionUtils::makeDescriptor(primDesc.weights_desc());
+    const auto originalWeightsDesc = MemoryDescUtils::convertToDnnlMemoryDesc(weiDesc);
+
+    // ignore the result since we just need to put the packed weights into the cache
+    (void)prepareWeightsMemory(originalWeightsDesc,
```

Comment:
Do I understand correclty that now we will prepack weights based on layout queried from dummy-shaped primitive descriptor (CreatePrimitve->preconfigure->create->DnnlFCExecutor->createShapeAgnosticData)?  It seems to be different with master where reorder happens on prepareParams stage only

Reply:
Oh, I missed `cacheWeights` parameter. So it is handled correclty than